.github/workflows/release-pypi.yaml

```diff
@@ -0,0 +1,36 @@
+name: Release Python Package
+
+on:
+  push:
+    branches:
+      - release/*
+
+jobs:
+  pre-release-check:
+    uses: cucumber/.github/.github/workflows/prerelease-checks.yml@main
+
+  test-python:
+    uses: ./.github/workflows/test-python.yml
+
+  publish-pypi:
+    name: Publish to PyPI
+    needs: [pre-release-check, test-python]
+    runs-on: ubuntu-latest
+    environment: Release
+    permissions:
+      id-token: write
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.8"
```

Comment:
Last minute check: Do we want to run the release scripts in 3.8 - @jsa34 just confirm this if you can